deploy/1-deploy.ts

```diff
@@ -27,11 +45,11 @@ const ADDRESSES: any = {
   },
 };
 
-module.exports = async ({ getNamedAccounts, deployments, network }: HardhatRuntimeEnvironment) => {
+const func: DeployFunction = async function ({ getNamedAccounts, deployments, network }: HardhatRuntimeEnvironment) {
   const { deploy } = deployments;
   const { deployer } = await getNamedAccounts();
 
-  const networkName = network.name;
+  const networkName = network.name as NetworkName;
 
   const { vBNBAddress, comptroller, WBNBAddress, timelock, acm } = ADDRESSES[networkName];
```

Comment:
```suggestion
  const  vBNBAddress = (await ethers.getContract('vBNB')).address
  const  comptrollerAddress = (await ethers.getContract('Comptroller')).address
  const  WBNBAddress = (await ethers.getContract('wBNB')).address
  const timelockAddress = (await ethers.getContract('NormalTimelock')).address
  const acmAddress = (await ethers.getContract('AccessControlManager')).address
```

Reply:
This will allow the deployed contracts to be fetched based on the network being used and will support hardhat. It will be up to the developer running this script on hardhat to make sure the scripts deploying these contracts runs first